muikku-rest/src/main/java/fi/otavanopisto/muikku/rest/user/UserRESTService.java

```diff
@@ -701,6 +702,54 @@ public Response updateFlag(@PathParam ("ID") Long id, fi.otavanopisto.muikku.res
 
     return Response.ok(createRestModel(flagController.updateFlag(flag, payload.getName(), payload.getColor(), payload.getDescription()))).build();
   }
+
+  @DELETE
+  @Path("/flags/{ID}")
+  @RESTPermit(handling = Handling.INLINE, requireLoggedIn = true)
+  public Response deleteFlag(@PathParam("ID") long flagId) {
+    Flag flag = flagController.findFlagById(flagId);
+
+    if (flag == null) {
+      return Response.status(Status.NOT_FOUND).build();
+    }
+    
+    boolean isOwner = false;
+    UserSchoolDataIdentifier ownerIdentifier = flag.getOwnerIdentifier();
+    SchoolDataIdentifier loggedIdentifier = sessionController.getLoggedUser();
+    if (loggedIdentifier == null) {
+      return Response.status(Status.BAD_REQUEST).entity("Must be logged in.").build();
+    }
+
+    UserSchoolDataIdentifier loggedUserIdentifier =
+        userSchoolDataIdentifierController.findUserSchoolDataIdentifierBySchoolDataIdentifier(
+            loggedIdentifier);
+    
+    if (loggedUserIdentifier == null) {
+      return Response
+                .status(Status.BAD_REQUEST)
+                .entity("No user school data identifier for logged user")
+                .build();
+    }
+    
+    if (Objects.equals(ownerIdentifier.getIdentifier(), loggedUserIdentifier.getIdentifier())) {
```

Comment:
On the other hand, in scripts owner is checked against userentity while here it's against identifier. Is this wanted functionality?